Alex's feedback

alexp616 opened the pull request:
Part 1/2 of my feedback:

I committed some typo fixes you guys can look through, and below are also some of my thoughts.

Generally, the font size feels too small - I looked at a bunch of other single-column papers and they all have bigger font size. This makes me feel like the information is a lot more condensed.

1.1 (Acknowledgements) doesn't seem to be finished, or is just formatted really weirdly

In Algorithm 2 (Splitting of Frobenius), the steps are a lot higher level than the pseudocode we give in other examples. Admittedly this is easier to understand, but should we try to standardize the "levelness" of our algorithm descriptions?

In Algorithm 6 (Merge algorithm), cmp is a vector, how are we comparing a vector to 0 in line 12? in line 20, the loop should probably go from l, ... , l + numLmatches - 1.

I agree that our algorithms don't look very pretty, I think we should try to look at some other papers where algorithms are described and take inspiration. For example in https://arxiv.org/pdf/0810.3203 (harvey cache friendly truncated fft), the algorithms on pages 5 and 8 aren't simple, but still manage to look more organized than ours

jjgarzella commented on algorithm_outline.tex:
```diff
@@ -663,7 +664,7 @@ \subsection{The key idea: finding the matrix of the linear operator ``multiply t
 
 \begin{algorithm}[H]
-\caption{Quasi-\(F\)-split height: matrix-based algorithm}
+\caption{quasi-\(F\)-split height: matrix-based algorithm}
```
When "quasi-F-split" is at the beginning of a sentence, it should be capitalized, I think this is standard.

jjgarzella commented:
With regards to cmp, it is supposed to be a vectorized comparison, at some point there was dot notation, but then the dot notation got taken out becasue that was confusing. But now it's confusing as it is because it seems to be comparing a vector to a scalar.

jjgarzella commented:
I think one thing that's awkward about our algorithms is that we are mixing CS notation with math notation. It seems like in most math papers, like the Harvey ones you mentioned, seem to use almost exclusively math notation.

I'm just not completely sure how the nicest way to write things like a vectorized compare in math notation.

Commit:
changed algorithm package

alexp616 committed:
Merge branch 'alex-edits-1' of https://github.com/jjgarzella/all-heights-k3-surfaces into alex-edits-1

alexp616 commented:
I changed the algorithm package for all of the algorithms, algorithms 1-5 I "restyled" and 6-7 I got too tired and just left them as is, I think they already look better than the old package because there's less empty space.

@jjgarzella Can you look at say, Algorithm 5 and Algorithm 6 side by side, and let me know whether I should keep going with the font changes? I think it makes the algorithms look better because it's not just a wall of text of the same font.

Also, I want to rename the monomial bases from $M_d$ to something else, so that we can name the resulting matrix $M$, which is what it's named in Algorithm 3

jjgarzella commented:
> I changed the algorithm package for all of the algorithms, algorithms 1-5 I "restyled" and 6-7 I got too tired and just left them as is, I think they already look better than the old package because there's less empty space.
> 
> @jjgarzella Can you look at say, Algorithm 5 and Algorithm 6 side by side, and let me know whether I should keep going with the font changes? I think it makes the algorithms look better because it's not just a wall of text of the same font.

I took a look, the monospace font makes it a million times better. Please keep going.

> 
> Also, I want to rename the monomial bases from M d to something else, so that we can name the resulting matrix M , which is what it's named in Algorithm 3

I'll suggest $B_d$, the "B" standing for "basis".

jjgarzella commented:
> In Algorithm 2 (Splitting of Frobenius), the steps are a lot higher level than the pseudocode we give in other examples. Admittedly this is easier to understand, but should we try to standardize the "levelness" of our algorithm descriptions?

I guess I don't really see the issue with high-levelness vs low-levelness? The algorithm pseudocode matches the code pretty closely...

One issue I see is that there are two for loops when there really should be one, that should make it simpler.

jjgarzella commented:
> In Algorithm 6 (Merge algorithm), cmp is a vector, how are we comparing a vector to 0 in line 12? in line 20, the loop should probably go from l, ... , l + numLmatches - 1.

Agreed

jjgarzella committed:
Small edits

Change `M` to `B_d`

Change the for loop in MERGE to be simpler

Refactor Algorithm 2

alexp616 committed:
polished all algorithms, added clarifying comments to merge

alexp616 committed:
fix line number references to old algorithms

alexp616 committed:
added edits for polynomial powering

alexp616 committed:
small edits for matmul

alexp616 committed:
small edits for heights

alexp616 committed:
Merge branch 'alex-edits-1' of https://github.com/jjgarzella/all-heights-k3-surfaces into alex-edits-1

alexp616 committed:
typo fix in algorithm 2

alexp616 commented:
@jjgarzella This is nearing the last of my edits. You can glance at all the commits, but the important one is the "polished all algorithms" commit. I didn't pull before making all these commits so I had to do a merge. I was pretty careful about it so I don't think I overwrote too much of your stuff. Big changes happened in algorithms 2 and 6 (splitting of Frobenius and Merge) - notably I added in-line comments in Merge, which I think are necessary since even I didn't understand Merge until I added those comments.

alexp616 committed:
fixed duplicate code from merge

jjgarzella committed:
Add timing numbers and discussion for p=11,13

jjgarzella committed:
Small edits

jjgarzella committed:
Fix typo

alexp616 committed:
final typo fixes

alexp616 committed:
Merge branch 'alex-edits-1' of https://github.com/jjgarzella/all-heights-k3-surfaces into alex-edits-1

alexp616 commented:
This branch ready to merge

alexp616 committed:
final typo fix v2
